File: iosApp/iosApp/Modules/MnemonicConfirmation ✅/MnemonicConfirmationViewModel.swift
// Created by web3d4v on 12/05/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

struct MnemonicConfirmationViewModel {
    let potentialWords: [String]
    let wordsInfo: [WordInfo]
    let isValid: Bool?
    let mnemonicToUpdate: String?
    let showSalt: Bool
    
    struct WordInfo {
        let word: String
        let isInvalid: Bool
    }
}

File: web3walletcore/src/commonMain/kotlin/com/sonsofcrypto/web3walletcore/modules/mnemonicConfirmation/MnemonicConfirmationViewModel.kt
package com.sonsofcrypto.web3walletcore.modules.mnemonicConfirmation

import com.sonsofcrypto.web3walletcore.common.helpers.MnemonicInputViewModel


data class MnemonicConfirmationViewModel(
    val potentialWords: List<String>,
    val wordsInfo: List<MnemonicInputViewModel.Word>,
    val isValid: Boolean?,
    val mnemonicToUpdate : String?,
    val showSalt: Boolean,
)
